packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -145,26 +146,32 @@ library LibProving {
         // Taiko's core protocol.
         {
             address verifier = resolver.resolve(tier.verifierName, true);
-            // The verifier can be address-zero, signifying that there are no
-            // proof checks for the tier. In practice, this only applies to
-            // optimistic proofs.
-            if (verifier == address(0) && tier.verifierName != TIER_OP) {
-                revert L1_MISSING_VERIFIER();
-            }
 
             if (verifier != address(0)) {
                 bool isContesting = proof.tier == ts.tier && tier.contestBond != 0;
 
                 IVerifier.Context memory ctx = IVerifier.Context({
                     metaHash: blk.metaHash,
                     blobHash: meta.blobHash,
+                    // TODO(Brecht): Quite limiting this is required to be the same address as
+                    // msg.sender,
+                    // less flexibility on the prover's side for proof generation/proof submission
+                    // using
+                    // multiple accounts.
+                    // Added msgSender to allow the prover to be any address in the future.
                     prover: msg.sender,
+                    msgSender: msg.sender,
```

Comment:
Added different field to allow the `prover` address to be more flexible, while not breaking verifiers that require a specific `msg.sender` to work (like the current guardian proof verifier). This is now explicit instead of it being implicit before that the prover was also required to be `msg.sender`.

Reply:
It seems we need to open a few PR to handle 'msg.sender' better.

Reply:
Currently yes! But I feel they should be separate, that's also one of the reasons we bake in the prover address in the proof so it cannot be stolen by someone else, so if somebody would like to submit someone's else's proof they should be able to do so, the prover address baked in the proof would always be seen as the actual prover to the protocol.